salt/minion.py

```diff
@@ -2591,7 +2591,8 @@ def _post_master_init(self, master):
 
         proxy_init_fn = self.proxy[fq_proxyname+'.init']
         proxy_init_fn(self.opts)
-        self.opts['grains'] = salt.loader.grains(self.opts)
+        self.opts['grains'] = salt.loader.grains(self.opts, force_refresh=True)
+        self.functions['saltutil.sync_grains'](saltenv='base')
```

Comment:
If you could add an in-line comment to the code explaining that, then I will merge this. Thanks. :]